dipy/reconst/dki.py

```diff
@@ -0,0 +1,1634 @@
+#!/usr/bin/python
+""" Classes and functions for fitting tensors """
+from __future__ import division, print_function, absolute_import
+
+import warnings
+
+import numpy as np
+
+import scipy.optimize as opt
+
+from dipy.reconst.dti import (TensorFit, fractional_anisotropy,
+                              geodesic_anisotropy, mean_diffusivity,
+                              axial_diffusivity, radial_diffusivity, trace,
+                              color_fa, determinant, isotropic, deviatoric,
+                              norm, mode, linearity, planarity, sphericity,
+                              apparent_diffusion_coef, from_lower_triangular,
+                              lower_triangular, decompose_tensor)
+
+from dipy.sims.voxel import DKI_signal
+from dipy.utils.six.moves import range
+from dipy.data import get_sphere
+from ..core.gradients import gradient_table
+from ..core.geometry import vector_norm
+from ..core.sphere import Sphere
+from .vec_val_sum import vec_val_vect
+from ..core.onetime import auto_attr
+from .base import ReconstModel
+
+
+def rdpython(x,y,z):
+    r"""
```

Comment:
Hi @garyfallidis. Thanks for all the comments!

The "WIP" markers here are to tell you that these functions are still work
in progress, and will change substantially before they are really ready for
review. For now, it would be good to focus on the code below line 435 or so
of this file (`def mean_kurtosis`). In particular, we are still working
through the math of the first few functions, and when we sort it all out,
there will also be clearer documentation here.

On Sat, Jun 27, 2015 at 1:31 PM, Eleftherios Garyfallidis <
notifications@github.com> wrote:

> In dipy/reconst/dki.py
> <https://github.com/nipy/dipy/pull/664#discussion_r33415901>:
>
> > +                              apparent_diffusion_coef, from_lower_triangular,
> > +                              lower_triangular, decompose_tensor)
> > +
> > +from dipy.sims.voxel import DKI_signal
> > +from dipy.utils.six.moves import range
> > +from dipy.data import get_sphere
> > +from ..core.gradients import gradient_table
> > +from ..core.geometry import vector_norm
> > +from ..core.sphere import Sphere
> > +from .vec_val_sum import vec_val_vect
> > +from ..core.onetime import auto_attr
> > +from .base import ReconstModel
> > +
> > +
> > +def rdpython(x,y,z):
> > +    r"""
>
> Hi @RafaelNH <https://github.com/RafaelNH> and @arokem
> <https://github.com/arokem>,
>
> Many of the functions need correction for PEP8 and the math equations will
> not render as they are in docstrings.
> Please you save some good time to correct these.
>
> Also in your functions you usually write WIP in the documentation. This is
> not a recommended practice. Try to write the documentation of your
> functions as you go. Otherwise you will end up with bad documentation
> (quickly written). As you write a function and you still have good memory
> of what it does it is easier to write the documentation. It makes it easier
> also for us to give you good reviews.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/664/files#r33415901>.
>